tests/unit/test_context.py

```diff
@@ -72,84 +88,91 @@ def test_explain():
-    assert_frame_equal(result.compute(), data_frame.compute())
+    assert isinstance(result, dd.DataFrame if not gpu else dask_cudf.DataFrame)
+    dd.assert_eq(result, data_frame)
 
     result = c.sql("SELECT * FROM df", return_futures=False)
-    assert isinstance(result, pd.DataFrame)
-    assert_frame_equal(result, data_frame.compute())
+    assert isinstance(result, pd.DataFrame if not gpu else cudf.DataFrame)
+    dd.assert_eq(result, data_frame)
 
+    if gpu:
+        data_frame = dask_cudf.from_dask_dataframe(data_frame)
     result = c.sql("SELECT * FROM other_df", dataframes={"other_df": data_frame})
-    assert isinstance(result, dd.DataFrame)
-    assert_frame_equal(result.compute(), data_frame.compute())
+    assert isinstance(result, dd.DataFrame if not gpu else dask_cudf.DataFrame)
+    dd.assert_eq(result, data_frame)
 
 
-def test_input_types(temporary_data_file):
+@pytest.mark.parametrize("gpu", [False, pytest.param(True, marks=pytest.mark.gpu)])
+def test_input_types(temporary_data_file, gpu):
     c = Context()
     df = pd.DataFrame({"a": [1, 2, 3]})
 
-    def assert_correct_output():
+    def assert_correct_output(gpu):
         result = c.sql("SELECT * FROM df")
-        assert isinstance(result, dd.DataFrame)
-        assert_frame_equal(result.compute(), df)
+        assert isinstance(result, dd.DataFrame if not gpu else dask_cudf.DataFrame)
+        dd.assert_eq(result, df)
 
-    c.create_table("df", df)
-    assert_correct_output()
+    c.create_table("df", df, gpu=gpu)
+    assert_correct_output(gpu=gpu)
 
-    c.create_table("df", dd.from_pandas(df, npartitions=1))
-    assert_correct_output()
+    c.create_table("df", dd.from_pandas(df, npartitions=1), gpu=gpu)
+    assert_correct_output(gpu=gpu)
 
     df.to_csv(temporary_data_file, index=False)
-    c.create_table("df", temporary_data_file)
-    assert_correct_output()
+    c.create_table("df", temporary_data_file, gpu=True)
+    assert_correct_output(gpu=gpu)
 
     df.to_csv(temporary_data_file, index=False)
-    c.create_table("df", temporary_data_file, format="csv")
-    assert_correct_output()
+    c.create_table("df", temporary_data_file, format="csv", gpu=gpu)
+    assert_correct_output(gpu=gpu)
 
     df.to_parquet(temporary_data_file, index=False)
-    c.create_table("df", temporary_data_file, format="parquet")
-    assert_correct_output()
+    c.create_table("df", temporary_data_file, format="parquet", gpu=gpu)
+    assert_correct_output(gpu=gpu)
 
     with pytest.raises(AttributeError):
-        c.create_table("df", temporary_data_file, format="unknown")
+        c.create_table("df", temporary_data_file, format="unknown", gpu=gpu)
 
     strangeThing = object()
 
     with pytest.raises(ValueError):
-        c.create_table("df", strangeThing)
+        c.create_table("df", strangeThing, gpu=gpu)
 
 
-def test_tables_from_stack():
+@pytest.mark.parametrize("gpu", [False, pytest.param(True, marks=pytest.mark.gpu)])
+def test_tables_from_stack(gpu):
```

Comment:
We should either add `cudf.DataFrames` to this check, or (ideally) generalize it for dataframe-like objects:

https://github.com/rapidsai/dask-sql/blob/58c58da7696ff01de2b7f3e1f27a51dba0261e7b/dask_sql/context.py#L784-L785